src/WikiParsing/Articles/Article.fs

```diff
@@ -29,25 +29,29 @@ let getCondition = function
     | Starts partName -> starts partName
     | OneOf partNames -> fun x -> partNames |> Seq.contains x
 
+let isReachable (url: string) = 
+    (new HttpClient()).GetAsync(url) 
```

Comment:
Is it possible to avoid generating a new HttpClient for reachability test?

Reply:
In the stateless world - I don't know. Do you have ideas?

Reply:
We could try using smth like this https://www.stevejgordon.co.uk/introduction-to-httpclientfactory-aspnetcore, but it will require either global IHttpClientFactory or doing some sort of containerization

Reply:
I asked the community, we got quite a [comprehensive answer](https://stackoverflow.com/a/59955102/3232646).

I don't know. I don't think anything global (e.g. client factory) in the program is a right move. So we could introduce IoC container. But yeah, for that we would need to revamp quite a big chunk here... I am reluctant to do it because it looks like a premature optimization in a way - we don't have a bottleneck here yet. 

What do you think?

Reply:
I agree, let's open an issue for that and merge the code

Reply:
Sure, [here](https://github.com/psfinaki/CheckYourCzech/issues/623) it is.